### PR TITLE
The real rest-api code review

### PR DESCRIPTION
**Pull request by Larspolo**
## Description
For this PR all the code needs to be reviewed. We already tested the frontend & backend, however didn't go through all the code. Have fun reading through 15k lines :)

### CODE REVIEW

**Comment by Larspolo** [comment]
Try not to update this branch. This will only further increase the amount of lines that have to be reviewed

**Comment by engelhamer** [comment]
What? This is the develop branch so I reckon there is no other option than tot update it?

File: src/django/VLE/serializers.py [hunk 1]
```diff
@@ -3,288 +3,328 @@
...
+#         'deadline': preset.deadline.strftime('%Y-%m-%d %H:%M'),
+#     }
+#
+#     if preset.type == Node.PROGRESS:
+#         result = {**base, **{'target': preset.target}}
+#     elif preset.type == Node.ENTRYDEADLINE:
+#         result = {**base, **{'template': template_to_dict(preset.forced_template)}}
+#
+#     return result
+#
```
Comment (BigDGW) [inline src/django/VLE/serializers.py]: Why is this left in as it is commented out?

File: src/django/test/test_get_apis.py [hunk 2]
```diff
@@ -320,32 +313,32 @@ def test_get_entrycomments(self):
...
+    #
+    #     login = test.logging_in(self, self.rein_user, self.rein_pass)
+    #     result = test.api_get_call(self, '/get_assignment_by_lti_id/12xy/', login).json()
+    #     self.assertEquals(result['assignment']['name'], 'Colloq')
+    #
+    #     # permissions and authorization check for the api call.
+    #     login = test.logging_in(self, self.no_perm_user, self.no_perm_pass)
+    #     test.api_get_call(self, '/get_assignment_by_lti_id/12xy/', login, status=403)
+    #     test.api_get_call(self, '/get_assignment_by_lti_id/random/', login, status=404)
+    #     test.test_unauthorized_api_get_call(self, '/get_assignment_by_lti_id/12xy/')
```
Comment (BigDGW) [inline src/django/test/test_get_apis.py]: Why is this left in?

File: src/vue/package-lock.json [hunk 3]
```diff
@@ -14879,6 +14889,20 @@
...
+    "vuex-persistedstate": {
+      "version": "2.5.4",
+      "resolved": "https://registry.npmjs.org/vuex-persistedstate/-/vuex-persistedstate-2.5.4.tgz",
+      "integrity": "sha512-XYJhKIwO+ZVlTaXyxKxnplrJ88Fnvk5aDw753bxzRw5/yMKLQ6lq9CDCBex2fwZaQcLibhtgJOxGCHjy9GLSlQ==",
+      "requires": {
+        "deepmerge": "2.1.1",
+        "shvl": "1.3.1"
+      }
+    },
     "ware": {
```
Comment (BigDGW) [inline src/vue/package-lock.json]: Why is this file not ignored?

File: src/vue/src/api/format.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    update (id, data = null) {
+        return auth.update('formats/' + id, data)
+            .then(response => response.data)
+    }
+
+    // create_template (name, fields) {
+    //     return auth.authenticatedPost('/create_template/', {name: name, fields: fields})
+    //         .then(response => response.data)
+    // }
+}
```
Comment (BigDGW) [inline src/vue/src/api/format.js]: Why is this left in?

File: src/vue/src/api/journal.js [hunk 5]
```diff
@@ -1,67 +1,44 @@
...
+    // /* Get assignment journals.
+    //  * Requests all the assignment journals.
+    //  * returns a list of all journals.
+    //  */
+    // update_publish_grades_assignment (aID, published) {
+    //     return auth.authenticatedPost('/update_publish_grades_assignment/', {aID: aID, published: published})
+    //         .then(response => response.data)
+    // },
+    //
 }
```
Comment (BigDGW) [inline src/vue/src/api/journal.js]: And why is this left in?

File: src/django/VLE/management/commands/preset_db.py [hunk 6]
```diff
@@ -67,59 +68,45 @@ def gen_users(self):
...
+            course = factory.make_course(c["name"], c["abbr"], c["start_date"], c["end_date"], author)
             role_teacher = Role.objects.get(name='Teacher', course=course)
             role_student = Role.objects.get(name='Student', course=course)
             for sid in c["students"]:
                 student = self.users[sid]
                 factory.make_participation(student, course, role_student)
             for cid in c["teachers"]:
-                if self.users[cid] == author:
+                if self.users[cid] == author:  # TODO why is no participation required if the user is author
                     continue
```
[the remark this answers is not in the document]
Comment (Larspolo) [reply src/django/VLE/management/commands/preset_db.py]: @mjvkeulen

[the remark this answers is not in the document]
Comment (BigDGW) [reply src/django/VLE/management/commands/preset_db.py]: Shouldnt this check be inside the make participation so we dont need repeated code?

**Comment by engelhamer** [comment]
This is getting out of hand.

**Pull request closed without merging**